### PR TITLE
Logs: Store queries that were run to get log results

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This was part of https://github.com/grafana/grafana/pull/33744. This PR stores the queries that were run to get those log results in logs model. This is useful for:
**1. Logs navigation** - we can check if queries match the ones that we are expecting and if not, refresh the navigation (currently we are using queries which I realised is buggy, cause if you start to type new query, navigation refreshes.)   
2. This is going to be great for **Download Logs** cause now we can show the queries that were run to get those logs. We couldn't do that before.

### CODE REVIEW

File: public/app/features/explore/Logs.tsx [hunk 1]
```diff
@@ -353,12 +353,12 @@ export class UnthemedLogs extends PureComponent<Props, State> {
           </div>
           <LogsNavigation
             logsSortOrder={logsSortOrder}
-            visibleRange={visibleRange}
+            visibleRange={visibleRange || absoluteRange}
```
Comment (gabor) [inline public/app/features/explore/Logs.tsx:356]:
```suggestion
            visibleRange={visibleRange ?? absoluteRange}
```

i recommend the `??` operator here, it explains better the i-want-to-use-absoluteRange-when-that-is-undefined (i did not test the change though)

File: public/app/features/explore/Logs.tsx [hunk 2]
```diff
@@ -353,12 +353,12 @@ export class UnthemedLogs extends PureComponent<Props, State> {
...
             absoluteRange={absoluteRange}
             timeZone={timeZone}
             onChangeTime={onChangeTime}
             loading={loading}
-            queries={queries}
+            queries={logsQueries || []}
```
Comment (gabor) [inline public/app/features/explore/Logs.tsx:361]:
```suggestion
            queries={logsQueries}
```

it seems to me that `logsQueries` can never be undefined based on https://github.com/grafana/grafana/blob/ivana/navigation-refresh/public/app/features/explore/Logs.tsx#L50 (change untested)

File: public/app/features/explore/LogsContainer.tsx [hunk 3]
```diff
@@ -133,7 +133,7 @@ export class LogsContainer extends PureComponent<PropsFromRedux & LogsContainerP
               width={width}
               getRowContext={this.getLogRowContext}
               getFieldLinks={this.getFieldLinks}
-              queries={queries}
+              logsQueries={logsQueries || []}
```
Comment (gabor) [inline public/app/features/explore/LogsContainer.tsx:136]:
```suggestion
              logsQueries={logsQueries ?? []}
```

i like `??` 😁  (same argument as the one above (untested change)

**Review by gabor — COMMENTED** [review]
change is ok 👍 i added some nitpicks.

File: public/app/features/explore/Logs.tsx (hunk 2 above)
Comment (ivanahuckova) [reply public/app/features/explore/Logs.tsx]: I decided to follow the pattern of other logs params and made it optional params. And therefore here, I have added `logsQueries ?? []`
https://github.com/grafana/grafana/pull/33971/commits/2fd0e8e9aabee04595dd865a50c9f157c3b90187

File: public/app/features/explore/LogsContainer.tsx (hunk 3 above)
Comment (ivanahuckova) [reply public/app/features/explore/LogsContainer.tsx]: This was changed to `logsQueries={logsQueries}` as logsQueries is now optional param.

**Review by gabor — APPROVED** [review]
looks good to me 👍

**Pull request merged**